### ISSUE TITLE
dotnet-svcutil.xmlserializer Assembly Load Error

### ISSUE THREAD

**Issue by bobcat1506** [dotnet/wcf#4090]
**Describe the bug**
The dotnet-svcutil.xmlserializer is encountering an error because it is trying to load an assembly that does not exist. C:\Users\Admin\.nuget\packages\system.private.servicemodel\4.7.0\ does not contain a runtimes folder.

**To Reproduce**
1. Follow the steps in this article https://docs.microsoft.com/en-us/dotnet/core/additional-tools/dotnet-svcutil.xmlserializer-guide

**Expected behavior**
Generation of a serialization assembly.

**Screenshots**
`Microsoft (R) Build Engine version 16.4.0+e901037fe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 23.16 ms for C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\SvcUtilXmlSerializePreGen.csproj.
  Restore completed in 59 ms for C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\SvcUtilXmlSerializePreGen.csproj.
  SvcUtilXmlSerializePreGen -> C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\bin\Debug\netstandard2.0\SvcUtilXmlSerializePreGen.dll
  Load Assembly From C:\Users\Admin\.nuget\packages\system.private.servicemodel\4.7.0\runtimes\win\lib\netstandard2.0\System.Private.ServiceModel.dll
EXEC : warning : Fail to load the assembly C:\Users\Admin\.nuget\packages\system.private.servicemodel\4.7.0\runtimes\win\lib\netstandard2.0\System.Private.ServiceModel.dll with the error Cannot load file C:\Users\Admin\.nuget\packages\system.private.servicemodel\4.7.0\runtimes\win\lib\netstandard2.0\System.Private.ServiceModel.dll as an Assembly. Check the FusionLogs for more Information. [C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\SvcUtilXmlSerializePreGen.csproj]

  Microsoft (R) dotnet-svcutil.xmlserializer tool, Version 1.1.0.
  [Microsoft (R) Windows (R) Communication Foundation]
  Copyright (c) Microsoft Corporation.  All rights reserved.

  If you would like more help, type "svcutil -?"
EXEC : warning : Cannot load file C:\Users\Admin\.nuget\packages\system.private.servicemodel\4.7.0\runtimes\win\lib\netstandard2.0\System.Private.ServiceModel.dll as an Assembly. Check the FusionLogs for more Information. [C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\SvcUtilXmlSerializePreGen.csproj]

      Could not load file or assembly 'C:\Users\Admin\.nuget\packages\system.private.servicemodel\4.7.0\runtimes\win\lib\netstandard2.0\System.Private.ServiceModel.dll'. The system cannot find the path specified.

C:\Users\Admin\.nuget\packages\dotnet-svcutil.xmlserializer\1.1.0\build\dotnet-svcutil.xmlserializer.targets(22,5): warning MSB3073: The command "dotnet svcutil.xmlserializer obj\Debug\netstandard2.0\SvcUtilXmlSerializePreGen.dll --quiet --out:obj\Debug\netstandard2.0\SvcUtilXmlSerializePreGen.XmlSerializers --smreference:"C:\Users\Admin\.nuget\packages\system.servicemodel.primitives\4.7.0\ref\netstandard2.0\System.ServiceModel.Primitives.dll"" exited with code 3. [C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\SvcUtilXmlSerializePreGen.csproj]
C:\Users\Admin\.nuget\packages\dotnet-svcutil.xmlserializer\1.1.0\build\dotnet-svcutil.xmlserializer.targets(23,5): warning : Warning : Fail to generate the serializer for SvcUtilXmlSerializePreGen.dll. [C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\SvcUtilXmlSerializePreGen.csproj]

Build succeeded.

EXEC : warning : Fail to load the assembly C:\Users\Admin\.nuget\packages\system.private.servicemodel\4.7.0\runtimes\win\lib\netstandard2.0\System.Private.ServiceModel.dll with the error Cannot load file C:\Users\Admin\.nuget\packages\system.private.servicemodel\4.7.0\runtimes\win\lib\netstandard2.0\System.Private.ServiceModel.dll as an Assembly. Check the FusionLogs for more Information. [C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\SvcUtilXmlSerializePreGen.csproj]
EXEC : warning : Cannot load file C:\Users\Admin\.nuget\packages\system.private.servicemodel\4.7.0\runtimes\win\lib\netstandard2.0\System.Private.ServiceModel.dll as an Assembly. Check the FusionLogs for more Information. [C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\SvcUtilXmlSerializePreGen.csproj]
C:\Users\Admin\.nuget\packages\dotnet-svcutil.xmlserializer\1.1.0\build\dotnet-svcutil.xmlserializer.targets(22,5): warning MSB3073: The command "dotnet svcutil.xmlserializer obj\Debug\netstandard2.0\SvcUtilXmlSerializePreGen.dll --quiet --out:obj\Debug\netstandard2.0\SvcUtilXmlSerializePreGen.XmlSerializers --smreference:"C:\Users\Admin\.nuget\packages\system.servicemodel.primitives\4.7.0\ref\netstandard2.0\System.ServiceModel.Primitives.dll"" exited with code 3. [C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\SvcUtilXmlSerializePreGen.csproj]
C:\Users\Admin\.nuget\packages\dotnet-svcutil.xmlserializer\1.1.0\build\dotnet-svcutil.xmlserializer.targets(23,5): warning : Warning : Fail to generate the serializer for SvcUtilXmlSerializePreGen.dll. [C:\Users\Admin\source\repos\SvcUtilXmlSerializePreGen\SvcUtilXmlSerializePreGen.csproj]
    4 Warning(s)
    0 Error(s)

Time Elapsed 00:00:02.22`

**Additional context**
Repo: https://github.com/bobcat1506/SvcUtilXmlSerializePreGen

**Comment by StephenBonikowsky** [comment]
@bobcat1506 Thanks for reporting the issue, we will look into it.

**Comment by bobcat1506** [comment]
Any update on this?  We have an app in production that is misbehaving and believe pre-generating the serialization assembly will resolve it.

**Comment by StephenBonikowsky** [comment]
Working on a fix right now, then need to get it packaged and published, shouldn't be too long before we have a fixed version up on NuGet.

**Comment by StephenBonikowsky** [comment]
Re-opening so we can use this issue to track package validation and publishing.

**Comment by StephenBonikowsky** [comment]
- [ ] build test package out of master and validate package
- [ ] cherry-pick fix from master to 3.1
- [ ] update package versioning and build release package
- [ ] final package validation
- [ ] publish new package to NuGet and delist previous packages.

**Comment by StephenBonikowsky** [comment]
@bobcat1506 Here is a private package of the tool which has the fix if you want to try it.
It will take a few days to produce a final release package and get it listed on NuGet.
[dotnet-svcutil.xmlserializer.1.2.0-preview1.20104.1.zip](https://github.com/dotnet/wcf/files/4167449/dotnet-svcutil.xmlserializer.1.2.0-preview1.20104.1.zip)

FYI, there is another issue to be aware of with this tool, see #4154

**Comment by bobcat1506** [comment]
Thanks @StephenBonikowsky. Both this issue and #4154 have impacted us significantly and frankly seem like pretty big misses from a regression testing perspective.  Will there be regression tests in the future to prevent these types of issue in future framework updates?  Any ETA on a fix for #4154?

**Comment by pmiddleton** [comment]
@StephenBonikowsky  - Any update on when the 1.2 package will hit nuget?  Are you waiting on a fix for #4154 first?

**Comment by StephenBonikowsky** [comment]
@pmiddleton We are proceeding with the final packaging changes, we can't wait on #4154. Assuming no complications we are shooting for a release to NuGet next week.

**Comment by bq1756** [comment]
Any complications?  Do you believe 1.2 will be released this week?

**Comment by StephenBonikowsky** [comment]
We had a build failure, infrastructure related, working to get it resolved and then final package validation before publishing the package.

**Comment by pmiddleton** [comment]
![image](https://user-images.githubusercontent.com/11600726/77954282-043c2c00-7294-11ea-928d-6ecee23232e6.png)

**Comment by StephenBonikowsky** [comment]
We got a good build and the package is going through some final validation steps and will then go to NuGet.

@dasetser will update and close this issue once it gets to NuGet.

**Comment by bobcat1506** [comment]
Any ETA on when final validation will be complete?

**Comment by StephenBonikowsky** [comment]
@dasetser @HongGit Is there an update on the package moving to NuGet?

**Comment by dasetser** [comment]
This was pushed to NuGet earlier today. You can find the new package at https://www.nuget.org/packages/dotnet-svcutil.xmlserializer/1.2.0

**Comment by bobcat1506** [comment]
I've updated the package reference and tried building it on Ubuntu targeting netcoreapp2.1, netcoreapp3.0 and netcoreapp3.1 and every time I get the same error:

Build succeeded.

/home/administrator/.nuget/packages/dotnet-svcutil.xmlserializer/1.2.0/build/dotnet-svcutil.xmlserializer.targets(35,5): warning MSB3073: The command "dotnet svcutil.xmlserializer obj/Debug/netcoreapp3.1/SvcUtilXmlSerializePreGen.dll --quiet --out:obj/Debug/netcoreapp3.1/SvcUtilXmlSerializePreGen.XmlSerializers --smreference:"/home/administrator/.nuget/packages/dotnet-svcutil.xmlserializer/1.2.0/lib/netcoreapp2.1/dotnet-svcutil.xmlserializer.dll;/home/administrator/.nuget/packages/system.private.servicemodel/4.7.0/lib/netstandard2.0/System.Private.ServiceModel.dll;/home/administrator/.nuget/packages/system.reflection.dispatchproxy/4.5.0/lib/netcoreapp2.0/System.Reflection.DispatchProxy.dll;/home/administrator/.nuget/packages/system.security.accesscontrol/4.5.0/lib/netstandard2.0/System.Security.AccessControl.dll;/home/administrator/.nuget/packages/system.security.cryptography.cng/4.5.0/lib/netcoreapp2.1/System.Security.Cryptography.Cng.dll;/home/administrator/.nuget/packages/system.security.cryptography.pkcs/4.5.0/lib/netcoreapp2.1/System.Security.Cryptography.Pkcs.dll;/home/administrator/.nuget/packages/system.security.cryptography.xml/4.5.0/lib/netstandard2.0/System.Security.Cryptography.Xml.dll;/home/administrator/.nuget/packages/system.security.permissions/4.5.0/lib/netstandard2.0/System.Security.Permissions.dll;/home/administrator/.nuget/packages/system.security.principal.windows/4.5.0/lib/netstandard2.0/System.Security.Principal.Windows.dll;/home/administrator/.nuget/packages/system.servicemodel.duplex/4.7.0/lib/netstandard2.0/System.ServiceModel.Duplex.dll;/home/administrator/.nuget/packages/system.servicemodel.http/4.7.0/lib/netstandard2.0/System.ServiceModel.Http.dll;/home/administrator/.nuget/packages/system.servicemodel.nettcp/4.7.0/lib/netstandard2.0/System.ServiceModel.NetTcp.dll;/home/administrator/.nuget/packages/system.servicemodel.primitives/4.7.0/lib/netcoreapp2.1/System.ServiceModel.Primitives.dll;/home/administrator/.nuget/packages/system.servicemodel.primitives/4.7.0/lib/netcoreapp2.1/System.ServiceModel.dll;/home/administrator/.nuget/packages/system.servicemodel.security/4.7.0/lib/netstandard2.0/System.ServiceModel.Security.dll"" exited with code 1. [/home/administrator/repos/github/SvcUtilXmlSerializePreGen/SvcUtilXmlSerializePreGen.csproj]
/home/administrator/.nuget/packages/dotnet-svcutil.xmlserializer/1.2.0/build/dotnet-svcutil.xmlserializer.targets(36,5): warning : Warning : Fail to generate the serializer for SvcUtilXmlSerializePreGen.dll. [/home/administrator/repos/github/SvcUtilXmlSerializePreGen/SvcUtilXmlSerializePreGen.csproj]
    2 Warning(s)
    0 Error(s)

**Comment by Oion71** [comment]
The same error on .Net 6.

\.nuget\packages\system.private.servicemodel\4.9.0\runtimes\win\lib\net6.0\System.Private.ServiceModel.dll as an Assembly. Check the FusionLogs for more Information.
